### PR TITLE
Integrates preliminary 404 search functionality

### PR DESCRIPTION
**Pull request by itsjoekent**
Fixes half of this https://github.com/DoSomething/dosomething/issues/3318

Pushing per request of @aaronschachter so it's easier for @sbsmith86 to start theming search results

### CODE REVIEW

File: lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module [hunk 1]
```diff
@@ -73,15 +73,37 @@ function dosomething_notfound_preprocess_node(&$vars)  {
...
-    $vars['search'] = false;
+  if (variable_get('dosomething_notfound_disable_results') == 1) {
     return;
   }
+
+  // @TODO: Check if its path specific (Intentionally hardcoded), get campaign suggestions, return & set suggestions var to true if it is 
+
+  $invalid_chars = array("/", "-", "_"); // request_path() shouldn't return a string with question marks and other chars we don't need
+  $raw_search = request_path();
+  $formmated_terms = explode(" ", str_replace($invalid_chars, " ", $raw_search));
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module]: Picky picky spelling.

File: lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module [hunk 2]
```diff
@@ -73,15 +73,37 @@ function dosomething_notfound_preprocess_node(&$vars)  {
...
   
-  //@TODO: Populate results info $vars['results'] = array();
-  //$vars['results_info'] = "We did a quick search, here is some similiar content that we do have.";
+  if (count($formmated_terms) <= 1) {
+    return;
+  }
+
+  $query_string = "";
+  $total = count($formmated_terms);
+  for($i = 1; $i < $total; $i++){ // Cannot start at 0, don't want that part of URL
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module]: Picky picky spacing.

**Comment by aaronschachter** [comment]
Looks good except for picky picky.  Feel free to fix in next PR. :+1:

**Comment by aaronschachter** [comment]
:+1: ship it value added to the value added

**Comment by itsjoekent** [comment]
more value added

**Comment by sbsmith86** [comment]
@deadlybutter Thanks for this! Looking to jump into the theming tomorrow.

**Comment by itsjoekent** [comment]
awesome

**Pull request merged**